pypeit/spectrographs/ntt_efosc2.py

```diff
@@ -256,4 +295,42 @@ def check_frame_type(self, ftype, fitstbl, exprng=None):
 
         msgs.warn('Cannot determine if frames are of type {0}.'.format(ftype))
         return np.zeros(len(fitstbl), dtype=bool)
+    
+    def bpm(self, filename, det, shape=None, msbias=None):
+        """
+        Generate a default bad-pixel mask.
+
+        Even though they are both optional, either the precise shape for
+        the image (``shape``) or an example file that can be read to get
+        the shape (``filename`` using :func:`get_image_shape`) *must* be
+        provided.
+
+        Args:
+            filename (:obj:`str` or None):
+                An example file to use to get the image shape.
+            det (:obj:`int`):
+                1-indexed detector number to use when getting the image
+                shape from the example file.
+            shape (tuple, optional):
+                Processed image shape
+                Required if filename is None
+                Ignored if filename is not None
+            msbias (`numpy.ndarray`_, optional):
+                Master bias frame used to identify bad pixels
+
+        Returns:
+            `numpy.ndarray`_: An integer array with a masked value set
+            to 1 and an unmasked value set to 0.  All values are set to
+            0.
+        """
+
+        # Call the base-class method to generate the empty bpm
+        bpm_img = super().bpm(filename, det, shape=shape, msbias=msbias)
+
+        msgs.info("Using hard-coded BPM for NTT EFOSC2")
+
+        bpm_img[182, 116:] = 1
```

Comment:
These positions are for 2x2 binning, may want to rewrite this to fit all the innings. 
The full BPM can be found at http://www.eso.org/sci/facilities/lasilla/instruments/efosc/inst/BADPIXMASK.html
Overall, the hard-coded bpm seems unnecessary for this instrument.

Reply:
Yes, we need to allow for different binnings.  For the moment, I would check what the binning is and if it is not 2x2 have it error out with a message "BPM code needs to be updated for your binning".

Reply:
Maybe remove it? Since it doesn't help with the overscan issue and the slit detection issue.